rfc/distributed-operation-cache.md

```diff
@@ -0,0 +1,107 @@
+---
+title: "Distributed Operation Cache"
+author: Dustin Deus
+date: 2024-08-25
+status: Draft
+---
+
+# Distributed Operation Cache
+
+- **Author:** Dustin Deus
+- **Date:** 2024-08-25
+- **Status:** Draft
+
+## Abstract
+
+This RFC describes a new feature to reduce the latency of the system by pre-planning the most expensive and requested operations before the router accepts traffic. We achieve this by computing the Top-N GraphQL operations available and making them available to all routers instances before they accept traffic.
+
+## Motivation
+
+GraphQL is a powerful tool to query data from a server. However, the flexibility of the query language comes with a cost. The cost is the complexity of the query and how expensive it is to normalize, plan and execute it. While execution performance is primarily a concern of the underlying subgraphs, the planning phase can be a unpredictable and significant latency contributor. The distributed operation cache aims to reduce this latency by pre-planning the most expensive and requested operations ahead to make it invisible to the user.
+
+# Proposal
```

Comment:
One general comment: Cache warming should allow for Operations to be incompatible with the client schema.
This could result in a warning, but should not raise an error.

Reply:
Done.